sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -258,6 +262,10 @@ private final class SemaphoreTaskInfo(val taskAttemptId: Long) extends Logging {
             // We now own the semaphore so we need to wake up all of the other tasks that are
             // waiting.
             hasSemaphore = true
+            if (trackSemaphore) {
+              nvtxRange =
+                Some(new NvtxUniqueRange(s"Sem-${taskAttemptId}", NvtxColor.ORANGE))
```

Comment:
`NvtxColor` is a hex number in argb (alpha, red, green, blue). I think it would make this a lot more useable if task X hashed to a specific color, consistently. That way you can tell them apart while the task is alive. Bonus points to make it color match with the current Task ranges? That might make this even more useful.

Reply:
This seems a really good idea. I will try as you suggested. Thanks!

Reply:
Actually changed my mind, and want to try it later. It is a nice idea and I definitely want to try, but don't want to keep this PR open for unnecessarily long. Created https://github.com/NVIDIA/spark-rapids/issues/11645.